e2e/single-cluster/bundle_diffs_test.go

```diff
@@ -43,20 +50,40 @@ var _ = Describe("BundleDiffs", func() {
 
 	BeforeEach(func() {
 		k = env.Kubectl.Namespace(env.Namespace)
+
+		var err error
+		targetNamespace, err = utils.NewNamespaceName()
+		Expect(err).ToNot(HaveOccurred())
+		name = "bundle-diff-" + targetNamespace
 	})
 
 	JustBeforeEach(func() {
-		out, err := k.Apply("-f", testenv.AssetPath(asset))
+		tmpdir, _ = os.MkdirTemp("", "fleet-")
+		gitrepo := path.Join(tmpdir, "gitrepo.yaml")
+		err := testenv.Template(gitrepo, testenv.AssetPath(asset), struct {
+			Name            string
+			TargetNamespace string
+		}{
+			name,
+			targetNamespace,
+		})
+		Expect(err).ToNot(HaveOccurred())
+
+		out, err := k.Apply("-f", gitrepo)
 		Expect(err).ToNot(HaveOccurred(), out)
 
 		DeferCleanup(func() {
-			out, err := k.Delete("-f", testenv.AssetPath(asset))
+			out, err := k.Delete("-f", gitrepo)
 			Expect(err).ToNot(HaveOccurred(), out)
 
+			os.RemoveAll(tmpdir)
+
+			k.Delete("namespace", targetNamespace)
```

Comment:
```suggestion
			_, _ = k.Delete("namespace", targetNamespace)
```

Reply:
Not sure what to do here, exiting from clean up because one step failed just leaves you with a mess... Maybe we should in the future log the error and continue.